### PR TITLE
VACMS-11230: Make Benefit Detail page unpublished alert blocks stand out

### LINKED ISSUE #11230: Unpublished Alert blocks are not visibly unpublished when selected on Benefits Detail Pages
## Description
Benefits Detail Pages have a `field_alert`, which allows an editor to place an Alert block at the top of the page.
> Alerts draw attention to information that may be critical to a veteran, family member, or caregiver. You can [create a new alert](https://staging.cms.va.gov/block/add/alert?destination=/admin/content/alerts), or reuse an existing one.

It is possible to create a situation where the selected Alert block is not published. This can be done by selecting an existing one that is unpublished or by creating a new one via the link in the instructions and leaving that in the default "draft" state.

While creating the Alert, it might seem quite obvious that the selected Alert is not published. However, after initial creation, that situation is much less obvious. It's not immediately clear via the node-edit screen that the selected Alert is not published, and this [has created some confusion](https://dsva.slack.com/archives/CDHBKAL9W/p1666030268937839?thread_ts=1665667127.482519&cid=CDHBKAL9W).

image

It also appears on the node Preview, despite it being unpublished:
image

It should be considered whether we might be able to indicate more clearly that a selected Alert is unpublished.

## Preferred fix
Style the Draft alert in the CMS UI using red highlighting like we do on draft nodes.

## Examples and links
Example BD node on prod: https://prod.cms.va.gov/node/16359/edit – "include alert" area/block displays a custom block that has been added to the page. If it were set to draft, it should appear pink.

## Acceptance Criteria
- [x] In node-edit screen for a Benefit Detail page, when an Alert block is selected:
  - [x] if the Alert is unpublished (never published or has unpublished changes), it appears with a light red background
  - [x] if the Alert is published, it appears with a white background
- [x] requires design review

### CMS Team
Please check the team(s) that will do this work.

- [ ] `CMS Team`
- [x] `Public Websites`
- [ ] `Facilities`
- [ ] `User support`

### PR DESCRIPTION
**Pull request by chri5tia**
## Description

Closes #11230 

- Distinguishes between published and unpublished alert blocks on the Benefit Details node view and form and in the entity browser. Benefit Details is the pilot.

## Testing done
Local functional testing:
- Multiple alert blocks (one via main content block)
- Placed published and unpublished alert block types
- Placed other unpublished alert block types
- Looked at alert blocks on other content types
- All of the above on both edit and view page

## Screenshots
Screenshot 2023-09-12 at 3 01 43 AM
Screenshot 2023-09-12 at 3 05 18 AM
Screenshot 2023-09-12 at 3 02 09 AM
Screenshot 2023-09-12 at 3 02 01 AM

## QA steps
As user 'testcontentadmin' with content admin privileges,
Go to a benefit details page, click to edit click to place an alert.
- [ ] Check that the entity browser has an option to select published true or false.
- [ ] Check that all the blocks in the list are published (status on the right of the table)
- [ ] Select a block and ensure that it does not show up highlighted in pink
- Remove the block now and place another one. This time select "False". 
- [ ] Check that the entity browser shows unpublished blocks
- [ ] Check that the draft blocks are highlighted in light red
- [ ] Select one of the draft alerts.
- [ ] Save the form. Check that the alert box is colored in light red.
- [ ] Ensure that the alert box on the view is also highlighted in light red.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/themes/custom/vagovclaro/vagovclaro.theme [hunk 1]
```diff
@@ -375,6 +376,38 @@ function vagovclaro_preprocess_node__documentation_page(&$variables) {
...
+ */
+function vagovclaro_preprocess_node__page(array &$variables): void {
+  // Get the page (Benefit Detail Page) node variable.
+  $node = $variables['node'];
+
+  // Initialize a variable to use in alert block twig template.
+  $variables['custom_alert_moderation_state_style'] = '';
+
+  // See if the Benefit Detail page has an alert.
+  if ($node->hasField('field_alert') && !$node->get('field_alert')->isEmpty()) {
```
Comment (chri5tia) [inline docroot/themes/custom/vagovclaro/vagovclaro.theme:390]: Restricting to the content type mentioned in the ticket. We should explore extending globally.

File: docroot/themes/custom/vagovclaro/vagovclaro.theme [hunk 2]
```diff
@@ -375,6 +376,38 @@ function vagovclaro_preprocess_node__documentation_page(&$variables) {
...
 
+/**
+ * Implements hook_preprocess_node__node_type().
+ */
+function vagovclaro_preprocess_node__page(array &$variables): void {
+  // Get the page (Benefit Detail Page) node variable.
+  $node = $variables['node'];
+
+  // Initialize a variable to use in alert block twig template.
+  $variables['custom_alert_moderation_state_style'] = '';
```
Comment (chri5tia) [inline docroot/themes/custom/vagovclaro/vagovclaro.theme:387]: This is leftover from a different strategy for passing a class to a twig template using this variable. The conditions of whether the block is published determines which class is passed to the variable to be rendered by the twig template. Should be removed if I don't move back to that strategy.

File: docroot/themes/custom/vagovclaro/vagovclaro.theme [hunk 3]
```diff
@@ -375,6 +376,38 @@ function vagovclaro_preprocess_node__documentation_page(&$variables) {
...
+      // Checks if block is of custom block type "alert".
+      if ($block_content instanceof BlockContentInterface && $block_content->bundle() == 'alert') {
+        // Get the alert block moderation state.
+        if ($block_content->hasField('moderation_state')
+          && !$block_content->get('moderation_state')->isEmpty()) {
+          $moderation_state = $block_content->get('moderation_state')->value;
+
+          // If the moderation state is draft, set a custom class.
+          if ($moderation_state == 'draft') {
+            $variables['content']['field_alert']['#attributes']['class'][] = 'node--unpublished';
```
Comment (chri5tia) [inline docroot/themes/custom/vagovclaro/vagovclaro.theme:402]: Alternatively, 

```
 // Set twig template variable conditionally.
          $variables['custom_alert_moderation_state_style'] = ($moderation_state == 'draft') ? 'custom-alert-draft' : 'custom-alert';
```

Comment (chri5tia) [reply docroot/themes/custom/vagovclaro/vagovclaro.theme:401]: Wait, I would be styling the `field_alert` entity reference field of `page` so the twig would have to be `field--node--field-alert--page.html.twig` I think

File: docroot/themes/custom/vagovclaro/vagovclaro.theme [hunk 4]
```diff
@@ -375,6 +376,37 @@ function vagovclaro_preprocess_node__documentation_page(&$variables) {
...
+ * Implements hook_preprocess_node__node_type().
+ */
+function vagovclaro_preprocess_node__page(array &$variables): void {
+  // Get the page (Benefit Detail Page) node variable.
+  $node = $variables['node'];
+
+  // See if the Benefit Detail page has an alert.
+  if ($node->hasField('field_alert') && !$node->get('field_alert')->isEmpty()) {
+    foreach ($node->get('field_alert')
+      ->referencedEntities() as $block_content) {
```
Comment (dsasser) [inline docroot/themes/custom/vagovclaro/vagovclaro.theme:389]: Suggestion:

I wouldn't wrap this line, it makes it harder to read.

File: docroot/themes/custom/vagovclaro/assets/scss/pages/_node-view.scss
Comment (chri5tia) [inline docroot/themes/custom/vagovclaro/assets/scss/pages/_node-view.scss:1]: I kind of want to move this to the module's css now.

File: docroot/modules/custom/va_gov_block_types/va_gov_block_types.module [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * Contains preprocess functions and other theme functions for the module.
+ */
+
+use Drupal\block_content\BlockContentInterface;
+use Drupal\block_content\Entity\BlockContent;
+
+/**
+ * Implements hook_preprocess_details().
+ */
+function va_gov_block_types_preprocess_details__node_page_edit_form__field_alert(array &$variables): void {
```
Comment (chri5tia) [inline docroot/modules/custom/va_gov_block_types/va_gov_block_types.module:14]: We went with hooks due to issues implementing the module needed for an entity browser to work with event subscriber but can make a ticket for it.

**Comment by chri5tia** [comment]
Found a bug where, now, when you place the alert block it isn't pink until you save it. Trying to figure out what changed in the code. Will have to explore this.

Screenshot 2023-09-12 at 3 20 49 AM

**Comment by chri5tia** [comment]
Not a bug. We didn't account for the behavior I was expecting this morning. Ready for PR review.

**Review by dsasser — APPROVED** [review]

File: docroot/modules/custom/va_gov_block_types/va_gov_block_types.module [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    return;
+  }
+
+  // Get the alert block status.
+  $block = $items['#block_content'];
+  if (!$block->hasField('status')) {
+    return 'No field block status';
+  }
+  $status = $block->get('status')->value;
+  echo $status;
```
Comment (chri5tia) [inline docroot/modules/custom/va_gov_block_types/va_gov_block_types.module:27]: Some debugging code was left in. Creating a follow-up PR to remove this.

**Pull request merged**